### ISSUE TITLE
Test Sanitize Overwrite

### ISSUE THREAD

**Issue by di98jgu** [sahlberg/libiscsi#202]
While it may or may not make sense there is nothing I know of 
in [SBC-3][1] [SBC-4][2] that prevent a thin provisioning 
unit to support Overwrite for sanitize.

Section 4.11.3 Completing a sanitize operation [SBC-3][1] 
[SBC-4][2] states that a unit is unmapped after a sanitize
commad have completed. 

My view is that this means that a unit with the LBPRZ flag
set in the read capacity 16 command, p. 144 table 66 [SBC-3][1],
and supports sanitize overwrite should return 0:s for all
LBA:s regardless of the pattern in the parameter list for the
sanitize command, p. 164 table 89 [SBC-3][1]. 

In other words the overrite service action will in this special case 
be identical to a block erase service action p. 163 table 88
[SBC-3][1]. 

The Sanitize Overwrite test fails in this case since the LBPRZ 
flag is not checked. I think the flag should be checked and the
expected response to a read command is all zeros.

[1]: "SCSI Block Commands - 3 (SBC-3) (12 May 2014) INCITS 514-2014"
[2]: "SCSI Block Commands – 4 (SBC-4) (01 Sep 2015) T10/BSR INCITS 506, Revision 8"

**Comment by sahlberg** [comment]
I have removed the warning if it is a SSD as there is no reason why an SSD can not support it.

But I don't think a device can silently substiture OVERWRITE for BLOCK ERASE.
The reasons are that OVERWRITE does not necessarily end up zeroing the block, therefore it can not just 'unmap' the blocks.
The second is that OVERWRITE is used to write alternating patterns multiple time over the physical medium so that any remaining traces or residuals of the original data is no longer detectable.
